File: src/integration-test/java/org/radarcns/redcap/testsuite/OrderedTestSuite.java
package org.radarcns.redcap.testsuite;

import org.junit.runner.RunWith;
import org.junit.runners.Suite;
import org.radarcns.redcap.EntryPointTest;
import org.radarcns.redcap.managementportal.MpClientTest;
import org.radarcns.redcap.integration.IntegratorTest;


/**
 * Test suite to ensure tests are run in a specific order so that various use-cases can be achieved.
 */
@Suite.SuiteClasses({MpClientTest.class, EntryPointTest.class, IntegratorTest.class})
@RunWith(Suite.class)
public class OrderedTestSuite {
}

File: src/integration-test/kotlin/org/radarcns/redcap/testsuite/OrderedTestSuite.kt
package org.radarcns.redcap.testsuite

import org.junit.runner.RunWith
import org.junit.runners.Suite
import org.junit.runners.Suite.SuiteClasses
import org.radarcns.redcap.EntryPointTest
import org.radarcns.redcap.integration.IntegratorTest
import org.radarcns.redcap.managementportal.MpClientTest

/**
 * Test suite to ensure tests are run in a specific order so that various use-cases can be achieved.
 */
@SuiteClasses(MpClientTest::class, EntryPointTest::class, IntegratorTest::class)
@RunWith(Suite::class)
class OrderedTestSuite